### PR TITLE
Nx test matrix speedup via multiprocessing, N = CPU count

### PR DESCRIPTION
**Pull request by mschwager**
Got tired of waiting for slow test matrix runs...

### CODE REVIEW

File: semgrep/semgrep/core_runner.py [hunk 1]
```diff
@@ -463,15 +461,9 @@ def handle_regex_patterns(
...
-            # Testing functionality runs in a multiprocessing.Pool. We cannot run
-            # a Pool inside a Pool, so we have to avoid multiprocessing when testing.
-            # https://stackoverflow.com/questions/6974695/python-process-pool-non-daemonic
-            matches = [get_re_matches(patterns_re, target) for target in targets]
-        else:
-            re_fn = functools.partial(get_re_matches, patterns_re)
-            with multiprocessing.Pool(self._jobs) as pool:
-                matches = pool.map(re_fn, targets)
+        re_fn = functools.partial(get_re_matches, patterns_re)
+        with pool.ThreadPool(self._jobs) as tpool:
```
Comment (mschwager) [inline semgrep/semgrep/core_runner.py:465]: This was originally a suggestion by @nbrahms. It allows us to more cleanly avoid this problem: https://stackoverflow.com/questions/6974695/python-process-pool-non-daemonic :+1:

**Comment by mschwager** [comment]
1m8s -> 17s on my machine :tada:

**Review by aryx — COMMENTED** [review]
nice!

**Review by colleend — APPROVED** [review]
awesome! gotta go fast 😎

**Review by minusworld — APPROVED** [review]

**Pull request merged**